Python_Engine/Compute/InstallReferencedVirtualenv.cs

```diff
@@ -0,0 +1,109 @@
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+
+namespace BH.Engine.Python
+{
+    public static partial class Compute
+    {
+        [Description("Install the Python Environment into BHoM for a referenced existing Python environment.")]
+        [Input("name", "The name of this BHoM Python Environment.")]
+        [Input("executable", "The Python executable for the referenced environment. From this, the version of Python will be determined in order to recreate it for BHoM as a virtualenv.")]
+        [Input("localPackage", "A local package to be included in the resultant environment. This is where custom BHoM code can be added into this environment.")]
+        [Input("run", "Run the installation process for the BHoM Python Environment.")]
+        [Output("env", "The virtualenv replication of  Python Environment for BHoM workflows.")]
+        public static oM.Python.PythonEnvironment InstallReferencedVirtualenv(
+            string name,
+            string executable, 
+            string localPackage = null, 
+            bool run = false
+        )
+        {
+            if (!IsValidEnvironmentName(name))
+            {
+                BH.Engine.Base.Compute.RecordError("A BHoM Python virtualenv cannot cannot contain invalid filepath characters.");
+                return null;
+            }
+
+            if (!File.Exists(executable))
+            {
+                BH.Engine.Base.Compute.RecordError($"{executable} doesn't exist. Have you installed LadybugTools Pollination?");
+                return null;
+            }
+
+            if (run)
+            {
```

Comment:
Please remove indented block. Perhaps consider adding a warning to alert user to toggle `run` to `true`.
```suggestion
            if (!run)
            {
               BH.Engine.Base.Compute.RecordWarning($"Please toggle `{nameof(run)}` to true.");
               return;
            }
```